lib/controller.js

```diff
@@ -641,6 +663,213 @@ controller.tesselFirmwareVerion = function(opts) {
+      if (label[0] !== 'E') {
+        callback(tessels[index].name, index, resolve, reject);
+        // controller.ssh.runSSH(index, opts, tessels, resolve, reject);
+      } else {
+        // going to clear screen and calling exit to resolve promise
+        controller.menu.clear();
+        controller.menu.exit(resolve);
+      }
+    });
+    //  });
+  },
+  makeMenu: function(tessel, index, rtm) {
+    if (tessel.lanConnection) {
+      if (!tessel.lanConnection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (not authorized) \n');
+      } else {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (authorized) \n');
+      }
+
+    } else if (tessel.usbConnection) {
+      if (!tessel.usbConnection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': [USB] (not authorized) \n');
+      } else {
+        rtm.add(index + ') ' + tessel.name + ': [USB] (authorized) \n');
+      }
+    }
+    // FIXME: Workaround for trouble with dynamic added menu elements
+    rtm._draw();
+  },
+  showMenu: function(rtm, resolve, reject) {
+    // Menu navigation
+    process.stdin.pipe(rtm.createStream()).pipe(process.stdout);
+
+    // raw mode for gain ability to navigate up and down within the menu
+    process.stdin.setRawMode(true);
+
+    // cb 
+    rtm.on('close', function() {
+      process.stdin.setRawMode(false);
+      process.stdin.end();
+    });
+    rtm.on('error', function(e) {
+      process.stdin.setRawMode(false);
+      reject(e);
+    });
+  },
+  seek: function(opts) {
+    // to be able to replace the seeker in testing mode
+    // return Tessel.seekTessels(opts);
+    return Tessel.get(opts);
+  },
+  clear: function() {
+    // selected exit!
+    process.stdout.write('\u001b[2J\u001b[0;0H');
+  },
+  exit: function(resolve) {
+    resolve();
+    //process.exit();
+  }
+};
+controller.ssh = {
+  runSSH: function(id, opts, tessels, resolve, reject) {
+    // clear console
+    debug('runSSH...\n root@' + tessels[id].lanConnection.host);
+    if (opts.menu) {
+      // because the test needs to resolve the promise ...
+      resolve({
+        opts: opts,
+        tessels: tessels
+      });
+    } else {
+      // if this is no test, starting the menu as child process
+      controller.menu.clear();
+      var ch = require('child_process')
+        .spawn('ssh', ['-i',
+          opts.path,
+          'root@' + tessels[id].lanConnection.ip
+        ], {
+          stdio: 'inherit'
+        });
+      logs.info('Connect to ' + tessels[id].lanConnection.host + '...');
+
+      // FIXME: There is no handler for poweroff Tessel ! Needs to be fixed within the firmware...
+      // (Terminal freezes imediatelly after poweroff the Tessel)
+
+      ch.once('error', function(e) {
+        if (e === 255) {
+          controller.menu.clear();
+          controller.ssh.notAuthorized();
```

Comment:
> @Student007 our model for T2 only involves one SSH key pair for all of your Tessels. I'm not sure what the point of having multiple keys would be.

@johnnyman727 I don't like this because it would be the same as not being able to change PASSWORDs... 
I vote for having `--key` options and review the code for necessary changes as I have done as first step in https://github.com/tessel/t2-cli/pull/330 
Also if you gave the key to someone or got hacked whatever ... **changing and having different keys for different Tessels** should be possible.
My two cents ...
Opened an issue for further discussion https://github.com/tessel/t2-cli/issues/347